micro-rdk/src/esp32/entry.rs

```diff
@@ -159,146 +177,204 @@ pub async fn serve_web_inner(
     srv.serve(robot).await;
 }
 
+async fn validate_robot_credentials(
+    exec: Esp32Executor,
+    robot_creds: &RobotCredentials,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let app_config = AppClientConfig::new(
+        robot_creds.robot_secret().to_owned(),
+        robot_creds.robot_id().to_owned(),
+        "".to_owned(),
+    );
+    let conn = Esp32Stream::TLSStream(Box::new(Esp32TLS::new_client().open_ssl_context(None)?));
+    let client = GrpcClient::new(conn, exec.clone(), "https://app.viam.com:443").await?;
+    let builder = AppClientBuilder::new(Box::new(client), app_config.clone());
+
+    let _client = builder.build().await?;
+    Ok(())
+}
+
+// Four cases:
+// 1) No Robot Credentials + WiFi without external network
+// 2) No Robot Credentials with external network\
+// 3) Robot Credentials with external network
+// 4) Robot Credentials + WiFi without external network
+
 #[cfg(feature = "provisioning")]
-async fn serve_provisioning_async<S>(
-    ip: std::net::Ipv4Addr,
+async fn serve_async<S>(
```

Comment:
I think this function is complex enough that it now warrants a general doc comment summarizing its procedure. I like that the cases have been outlined, but a quick statement on the expected behavior with each case would clarify things. I feel this way about `serve_async_with_external_network` as well.